File: service-ai/src/main/java/org/example/serviceai/entry/UserAiConfig.java
package org.example.serviceai.entry;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.example.serviceai.dto.UserAiConfigDto;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName(value = "user_ai_config", autoResultMap = true)
public class UserAiConfig {
    @TableId(type = IdType.AUTO)
    private Long userAiConfigId;
    private Long userId;
    private String groupName;
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<String> modelNames;
    private String aiUrl;
    private String apiKey;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
    public UserAiConfig (UserAiConfigDto userAiConfigDto) {
        this.groupName = userAiConfigDto.getGroupName();
        this.modelNames = userAiConfigDto.getModelNames();
        this.aiUrl = userAiConfigDto.getAiUrl();
        this.createTime = LocalDateTime.now();
        this.updateTime = LocalDateTime.now();
    }
}

File: codewise-agent/entry/UserAiConfig.py
from datetime import datetime
from typing import List

from sqlalchemy import DateTime, BigInteger, JSON, String
from sqlalchemy.orm import Mapped, mapped_column

from entry.database import Base


class UserAiConfig(Base):
    __tablename__ = "user_ai_config"

    user_ai_config_id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(BigInteger, nullable=False, index=True)
    group_name: Mapped[str] = mapped_column(String(255), nullable=False)
    model_names: Mapped[List[str]] = mapped_column(JSON, nullable=False, default=list)
    ai_url: Mapped[str] = mapped_column(String(512), nullable=False)
    api_key: Mapped[str] = mapped_column(String(512), nullable=False)
    create_time: Mapped[datetime] = mapped_column(DateTime, nullable=False, default=datetime.utcnow)
    update_time: Mapped[datetime] = mapped_column(
        DateTime,
        nullable=False,
        default=datetime.utcnow,
        onupdate=datetime.utcnow,
    )
